crates/kitsune_p2p/dht_arc/src/dht_arc.rs

```diff
@@ -300,6 +300,32 @@ impl DhtArcRange<DhtLocation> {
         matches!(self, Self::Bounded(_, _))
     }
 
+    /// Get the min distance to a location.
+    /// Zero if Full, u32::MAX if Empty.
+    pub fn dist(&self, tgt: u32) -> u32 {
+        match self {
+            DhtArcRange::Empty => u32::MAX,
+            DhtArcRange::Full => 0,
+            DhtArcRange::Bounded(start, end) => {
+                let start = u32::from(*start);
+                let end = u32::from(*end);
+                if start < end {
+                    if tgt >= start && tgt <= end {
+                        0
+                    } else if tgt < start {
+                        std::cmp::min(start - tgt, (u32::MAX - end) + tgt)
+                    } else {
+                        std::cmp::min(tgt - end, (u32::MAX - tgt) + start)
+                    }
+                } else if tgt <= end || tgt >= start {
+                    0
+                } else {
+                    std::cmp::min(tgt - end, start - tgt)
```

Comment:
Not that it makes a significant difference but I think this is off by 1? And the inequalities should still prevent overflow.

```suggestion
                        std::cmp::min(start - tgt, (u32::MAX - end) + tgt + 1)
                    } else {
                        std::cmp::min(tgt - end, (u32::MAX - tgt) + start + 1)
                    }
                } else if tgt <= end || tgt >= start {
                    0
                } else {
                    std::cmp::min(tgt - end, start - tgt)
```

Reply:
@maackle - this is a direct port of what the sql was doing. The existing unit test uses this exact logic to validate it. Do you think I should still change it?

Reply:
Well if you agree that it's correct, then yeah, this is a good chance to fix it. Maybe the tests will still pass?

Reply:
Okay, figuring this out again. Using base-5, so 0,1,2,3,4:

*with* the +1
- start: 2, end: 3, tgt: 1 = min(2 - 1, (4 - 3) + 1 + 1) = min(1, 3) = I *think* this is correct
- start: 2, end: 3, tgt: 4 = min(4 - 3, (4 - 4) + 2 + 1) = min(1, 3) = I *think* this is correct

a'ight, cool, will update :+1: